memory_745_747_755_757.x

```diff
@@ -0,0 +1,131 @@
+  /*   Suitable for use as DMA buffers.                                      */
+  /* - SRAM4 is in D3 and additionally accessible by the BDMA. Used for BDMA */
+  /*   buffers, for storing application data in lower-power modes, or for    */
+  /*   sharing data between the Cortex-M4 and Cortex-M7 cores.               */
+  /* - Zero wait-states.                                                     */
+  SRAM1   : ORIGIN = 0x30000000, LENGTH = 128K
+  SRAM2   : ORIGIN = 0x30020000, LENGTH = 128K
+  SRAM3   : ORIGIN = 0x30040000, LENGTH = 32K
+  SRAM4   : ORIGIN = 0x38000000, LENGTH = 64K
+
+  /* Backup SRAM */
+  /* Used to store data during low-power sleeps. */
+  BSRAM   : ORIGIN = 0x38800000, LENGTH = 4K  
+}
+
+/*
+/* Assign the memory regions defined above for use. */
+/*
+
+/* Provide the mandatory FLASH and RAM definitions for cortex-m-rt's linker script. */
+/* - These definitions concern only the CM7 core (CPU1).                            */
+/* - The CM4 core (CPU2) is set-up separately in the various other sections, below. */
+REGION_ALIAS(FLASH, FLASH1);
+REGION_ALIAS(RAM,   DTCM);
+
+/* The location of the stack can be overridden using the `_stack_start` symbol.           */
+/* - Set the CM7 core (CPU1) stack location at the end of RAM, using all remaining space. */
+/* - Set the CM4 core (CPU2) stack location in SRAM2, using the whole memory bank.        */
+_stack_start      = ORIGIN(RAM)   + LENGTH(RAM);
+_cpu2_stack_start = ORIGIN(SRAM2) + LENGTH(SRAM2);
+
+/* The location of the .text section can be overridden using the  */
+/* `_stext` symbol. By default it will place after .vector_table. */
+/* _stext = ORIGIN(FLASH) + 0x40c; */
+
+/* Define sections for placing symbols into the extra memory regions above.   */
+/* This makes them accessible from code.                                      */
+/* - ITCM, DTCM and AXISRAM connect to a 64-bit wide bus -> align to 8 bytes. */
+/* - All other memories     connect to a 32-bit wide bus -> align to 4 bytes. */
+/* - The SRAM1 and SRAM2 sections are commonly used as the stack and heap for */
+/*   the CM4 core in dual core versions and may thus have to be omitted from  */
+/*   this list.                                                               */
+SECTIONS {
+  .flash2 : ALIGN(4) {
+    LONG(_cpu2_stack_start);
+    KEEP(*(.flash2.reset_vector));
+    KEEP(*(.flash2.vector_table));
+    *(.flash2 .flash2.*);
+    . = ALIGN(4);
+    } > FLASH2
+
+  /*.itcm (NOLOAD) : ALIGN(8) { */
+  /*  *(.itcm .itcm.*);         */
+  /*  . = ALIGN(8);             */
+  /*  } > ITCM                  */
+    
+  .itcm : ALIGN(4) {
+    *(.itcm .itcm.*);
+    . = ALIGN(4);
+    } > ITCM AT > FLASH
+    
+  .axisram (NOLOAD) : ALIGN(8) {
+    *(.axisram .axisram.*);
+    . = ALIGN(8);
+    } > AXISRAM
+    
+  .sram1 (NOLOAD) : ALIGN(4) {
+    *(.sram1 .sram1.*);
+    . = ALIGN(4);
+    } > SRAM1
+    
+  .sram2 (NOLOAD) : ALIGN(4) {
```

Comment:
I think it's worth another note here that sram2 is used as the location for the CPU2 stack